pkg/chunk/aws_storage_client.go

```diff
@@ -382,22 +621,67 @@ func (b dynamoDBWriteBatch) Add(tableName, hashValue string, rangeValue []byte,
 	})
 }
 
-type dynamoDBReadBatch []map[string]*dynamodb.AttributeValue
-
-func (b dynamoDBReadBatch) Len() int {
-	return len(b)
+// Fill 'to' with WriteRequests from 'from' until 'to' has at most max requests. Remove those requests from 'from'.
+func (b dynamoDBWriteBatch) takeReqs(from dynamoDBWriteBatch, max int) {
```

Comment:
`to` isn't defined.

Also, why is this private when `Len` and `Add` are public?